### PR TITLE
feature: vgg-16 with mkldnn backend

### PR DESCRIPTION
**Pull request by i8run**
## What changes were proposed in this pull request?

This patch add vgg-16 on imagenet example.

## How was this patch tested?

Tested on 32-node bdw cluster.

### CODE REVIEW

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/InitializationMethod.scala [hunk 1]
```diff
@@ -278,12 +278,23 @@ case class ConstInitMethod(value: Double) extends InitializationMethod {
  *  (http://jmlr.org/proceedings/papers/v9/glorot10a/glorot10a.pdf)
  */
 case object Xavier extends InitializationMethod {
+  private var version2: Boolean = false
+
+  def setVersion2(v: Boolean): this.type = {
+    version2 = v
```
Comment (jason-dai) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/InitializationMethod.scala]: when should one use version2?

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/optim/SGD.scala [hunk 2]
```diff
@@ -306,7 +306,7 @@ object SGD {
     override def updateHyperParameter[T](optimMethod: SGD[T]): Unit = {
       val nevals = optimMethod.state.get[Int]("evalCounter").getOrElse(0)
       val lr = optimMethod.learningRate
-      val polyIter = nevals - excludeIterations
+      val polyIter = nevals // fix: should have no exclude iterations.
```
Comment (jason-dai) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/optim/SGD.scala]: this is no longer consistent with line#295

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/InitializationMethod.scala (hunk 1 above)
Comment (i8run) [reply spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/InitializationMethod.scala]: If the model is defined by dnn backend, we should use version2, such as the vgg16 in the example.

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/models/vgg/TrainImageNet.scala [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+          nesterov = param.nesterov, learningRateSchedule = lrSchedules)
+      }
+
+      val logdir = "vgg16-imagenet"
+      val appName = s"${sc.applicationId}"
+      val trainSummary = TrainSummary(logdir, appName)
+      trainSummary.setSummaryTrigger("LearningRate", Trigger.severalIteration(1))
+      trainSummary.setSummaryTrigger("Parameters", Trigger.severalIteration(10))
+
+      val criterion = Engine.getEngineType() match {
```
Comment (wzhongyuan) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/models/vgg/TrainImageNet.scala]: why different engines need different criterion functions?  do they have different topologies?

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/InitializationMethod.scala (hunk 1 above)
Comment (wzhongyuan) [reply spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/InitializationMethod.scala]: version2 seems quite confusing, can we have a meaningful name?

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/models/vgg/TrainImageNet.scala (hunk 3 above)
Comment (i8run) [reply spark/dl/src/main/scala/com/intel/analytics/bigdl/models/vgg/TrainImageNet.scala]: Yes. They have different topologies.

**Comment by wzhongyuan** [comment]
LGTM

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/optim/SGD.scala (hunk 2 above)
Comment (qiuxin2012) [reply spark/dl/src/main/scala/com/intel/analytics/bigdl/optim/SGD.scala]: @i8run  This change lead the learningRate of TrainInceptionV1 inconsistent, if warmupEpoch and maxLR is set.

**Pull request merged**